packages/api/portfolioDrafts/submit/submitPortfolioDraft.ts

```diff
@@ -0,0 +1,104 @@
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  DynamoDBDocumentClient,
+  UpdateCommand,
+  UpdateCommandOutput,
+  GetCommand,
+  GetCommandOutput,
+} from "@aws-sdk/lib-dynamodb";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import { isBodyPresent, isPathParameterPresent } from "../../utils/validation";
+import {
+  DATABASE_ERROR,
+  NO_SUCH_PORTFOLIO_DRAFT,
+  REQUEST_BODY_NOT_EMPTY,
+  PORTFOLIO_ALREADY_SUBMITTED,
+  PATH_PARAMETER_REQUIRED_BUT_MISSING,
+} from "../../utils/errors";
+import { PortfolioDraft } from "../../models/PortfolioDraft";
+import { v4 as uuidv4 } from "uuid";
+import { ProvisioningStatus } from "../../models/ProvisioningStatus";
+const TABLE_NAME = process.env.ATAT_TABLE_NAME ?? "";
+
+/**
+ * Submits all progress from the Portfolio Provisioning Wizard
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
+  if (isBodyPresent(event.body)) {
+    return REQUEST_BODY_NOT_EMPTY;
+  }
+
+  const portfolioDraftId = event.pathParameters?.portfolioDraftId;
+
+  if (!isPathParameterPresent(portfolioDraftId)) {
+    return PATH_PARAMETER_REQUIRED_BUT_MISSING;
+  }
+
+  try {
+    const result = await submitPortfolioDraftCommand(TABLE_NAME, portfolioDraftId);
+    return new ApiSuccessResponse(result.Attributes as PortfolioDraft, SuccessStatusCode.ACCEPTED);
+  } catch (error) {
+    console.log("Error: " + error);
+    // Check if the portfolioDraft exists
+    if (error.name === "ConditionalCheckFailedException") {
+      const getResult = await doesPortfolioDraftExistCommand(TABLE_NAME, portfolioDraftId);
+      console.log("getResult: " + JSON.stringify(getResult));
+      if (!getResult.Item) {
+        return NO_SUCH_PORTFOLIO_DRAFT;
+      }
+      return PORTFOLIO_ALREADY_SUBMITTED;
+    }
+    console.log("Database error: " + error);
+    return DATABASE_ERROR;
+  }
+}
+export async function submitPortfolioDraftCommand(
+  table: string,
+  portfolioDraftId: string
+): Promise<UpdateCommandOutput> {
+  const dynamodb = new DynamoDBClient({});
+  const ddb = DynamoDBDocumentClient.from(dynamodb);
```

Comment:
Here, and elsewhere in this file, re-use the stuff in `../utils/dynamodb.ts` or justify why we can't. You should `import { dynamodbDocumentClient as client } from "../utils/dynamodb"`

Reply:
And to clarify why I'm hitting to hard on reusing that, we're going to need it when we start actually setting up private networking and specifying actual attributes to pass to the constructor of `DynamoDBClient`. I'd far rather do that in one place than 100 and we need to understand why that code is not being used today.

Reply:
@kylelaker-ccpo Good catch.  I've not been on the lookout for this.